czkawka_core/src/duplicate.rs

```diff
@@ -1314,16 +1347,23 @@ fn delete_files(vector: &[FileEntry], delete_method: &DeleteMethod, warnings: &m
                     q_index = index;
                 }
             }
-            let src = vector[q_index].path.clone();
+            let src = &vector[q_index].path;
             for (index, file) in vector.iter().enumerate() {
-                if q_index != index {
-                    if let Err(e) = make_hard_link(&src, &file.path) {
-                        failed_to_remove_files += 1;
-                        warnings.push(format!("Failed to link {} -> {} ({})", file.path.display(), src.display(), e));
-                    } else {
-                        removed_files += 1;
-                        gained_space += file.size;
-                    }
+                if q_index == index {
+                    continue;
+                }
+                let r = if dryrun {
+                    warnings.push(format!("Replace file {} with hard link to {}", file.path.display(), src.display()));
```

Comment:
`warnings` should be changed to `messages`.
Errors means non recoverable errors like e.g. not providing valid directories
Warnings informs about e.g. failing in remove, link, add some objects
Messages just are informational

Reply:
Makes sense.